geth/account/accounts.go

```diff
@@ -14,30 +13,32 @@ import (
 	gethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/status-im/status-go/extkeys"
-	"github.com/status-im/status-go/geth/common"
-	"github.com/status-im/status-go/geth/rpc"
 )
 
 // errors
 var (
-	ErrAddressToAccountMappingFailure  = errors.New("cannot retrieve a valid account for a given address")
-	ErrAccountToKeyMappingFailure      = errors.New("cannot retrieve a valid key for a given account")
-	ErrWhisperIdentityInjectionFailure = errors.New("failed to inject identity into Whisper")
-	ErrWhisperClearIdentitiesFailure   = errors.New("failed to clear whisper identities")
-	ErrNoAccountSelected               = errors.New("no account has been selected, please login")
-	ErrInvalidMasterKeyCreated         = errors.New("can not create master extended key")
+	ErrAddressToAccountMappingFailure = errors.New("cannot retrieve a valid account for a given address")
+	ErrAccountToKeyMappingFailure     = errors.New("cannot retrieve a valid key for a given account")
+	ErrNoAccountSelected              = errors.New("no account has been selected, please login")
+	ErrInvalidMasterKeyCreated        = errors.New("can not create master extended key")
 )
 
-// Manager represents account manager interface
+// GethServiceProvider provides required geth services.
+type GethServiceProvider interface {
+	AccountManager() (*accounts.Manager, error)
+	AccountKeyStore() (*keystore.KeyStore, error)
+}
+
+// Manager represents account manager interface.
 type Manager struct {
-	nodeManager     common.NodeManager
-	selectedAccount *common.SelectedExtKey // account that was processed during the last call to SelectAccount()
+	geth            GethServiceProvider
+	selectedAccount *SelectedExtKey // account that was processed during the last call to SelectAccount()
 }
 
-// NewManager returns new node account manager
-func NewManager(nodeManager common.NodeManager) *Manager {
+// NewManager returns new node account manager.
+func NewManager(geth GethServiceProvider) *Manager {
```

Comment:
frankly speaking doesn't make much sense to me to name this thing as geth

Reply:
that name comes from a previous [discussion](https://github.com/status-im/status-go/pull/742#discussion-diff-175408867R39), can you think on a better naming for this?

Reply:
GethStuff ?))

Reply:
if serious, these are both objects from geth/accouns package. idk the good name for them

Reply:
@dshulyak  disagreement time :-P
It does totally make sense to name it as `geth` to improve convenience and to notice weird patters on-the-fly.

Sure, it is not impossible to figure out that `*accounts.Manager` return value is actually not `*account.Manager`, because the first one (by imports) is from `geth` and the second one is from our codebase.
If you see a call `m.geth.AccountManager().DoBlah()` you kind of get an idea about what is being returned.
If you see a call `m.provider.AccountManager().DoBlah()` you have much harder time reading this code.